src/lazy/expanded/e_expression.rs

```diff
@@ -123,8 +123,12 @@ impl<'top, D: Decoder> EExpression<'top, D> {
             MacroKind::MakeSymbol => {
                 MacroExpansionKind::MakeSymbol(MakeTextExpansion::symbol_maker(arguments))
             }
-            MacroKind::MakeSExp => MacroExpansionKind::MakeSExp(MakeSExpExpansion::new(arguments)),
             MacroKind::Annotate => MacroExpansionKind::Annotate(AnnotateExpansion::new(arguments)),
+            MacroKind::Flatten => MacroExpansionKind::Flatten(FlattenExpansion::new(
+                self.context,
+                environment,
+                arguments,
+            )),
```

Comment:
🪧 Now that `make_sexp` is a trivial template that uses `flatten`, we no longer need a special `MacroKind` and custom impl for it. While this PR adds 3 new system macros, `MacroKind`'s number of variants is unchanged.